opampserver/pkg/server/handlers.go

```diff
@@ -50,56 +55,73 @@ func (c *ConnectionHandlers) OnNewConnection(ctx context.Context, deviceId strin
 		return nil, nil, fmt.Errorf("missing pid in agent description")
 	}
 
+	if firstMessage.PackageStatuses == nil {
+		// this might come from an older agent which not yet supports package statuses
+		c.logger.Info("missing package statuses in first agent to server message", "deviceId", deviceId)
+	}
+
 	k8sAttributes, pod, err := c.deviceIdCache.GetAttributesFromDevice(ctx, deviceId)
 	if err != nil {
 		c.logger.Error(err, "failed to get attributes from device", "deviceId", deviceId)
 		return nil, nil, err
 	}
-	serverOfferedResourceAttributes, err := calculateServerAttributes(k8sAttributes, c.nodeName)
-	if err != nil {
-		c.logger.Error(err, "failed to calculate server attributes", "deviceId", deviceId)
-		return nil, nil, err
+
+	podWorkload := common.PodWorkload{
+		Namespace: pod.GetNamespace(),
+		Kind:      k8sAttributes.WorkloadKind,
+		Name:      k8sAttributes.WorkloadName,
 	}
 
 	instrumentedAppName := workload.GetRuntimeObjectName(k8sAttributes.WorkloadName, k8sAttributes.WorkloadKind)
-	c.logger.Info("new OpAMP client connected", "deviceId", deviceId, "namespace", k8sAttributes.Namespace, "podName", k8sAttributes.PodName, "instrumentedAppName", instrumentedAppName, "workloadKind", k8sAttributes.WorkloadKind, "workloadName", k8sAttributes.WorkloadName, "containerName", k8sAttributes.ContainerName, "otelServiceName", k8sAttributes.OtelServiceName)
+	remoteResourceAttributes, err := configresolvers.CalculateServerAttributes(k8sAttributes, c.nodeName)
+	if err != nil {
+		c.logger.Error(err, "failed to calculate server attributes", "k8sAttributes", k8sAttributes)
+		return nil, nil, err
+	}
 
-	sdkConfig, err := json.Marshal(RemoteConfigSdk{RemoteResourceAttributes: serverOfferedResourceAttributes})
+	fullRemoteConfig, err := c.sdkConfig.GetFullConfig(ctx, remoteResourceAttributes, &podWorkload)
```

Comment:
We can maybe send the instrumentedAppName to the GetFullConfig so we wont need to call GetRuntimeObjectName again in it?

Reply:
great catch. fixed 👍